### PR TITLE
Globs: Change Glob-not-found warning

### PR DESCRIPTION
**Pull request by mrtes**
The new message suggests users to remove unmatched files from their
.coafile. This feedback is more explicit and  can help to avoid user
confusion.

Closes https://github.com/coala/coala/issues/4146

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by mrtes** [comment]
I just noticed that some parts of the test-case I have modified got reformatted in the process.
This is likely due to auto-formatting in my editor (using autopep8). 
Should I amend this/remove the plugin?

**Comment by SanketDG** [comment]
@mrtes, yes please remove the autofixing. The code is readable enough as it is.

File: coalib/collecting/Collectors.py [hunk 1]
```diff
@@ -82,7 +82,9 @@ def collect_files(file_paths, log_printer, ignored_file_paths=None,
         collected_files, file_globs_with_files = [], []
 
     _warn_if_unused_glob(log_printer, file_paths, file_globs_with_files,
-                         "No files matching '{}' were found.")
+                         "No files matching '{}' were found. "
+                         'If this rule is not required, you can remove it '
```
Comment (SanketDG) [inline coalib/collecting/Collectors.py]: Can you maintain uniformity and use either double or single quotes? Double looks good here.

Comment (mrtes) [inline coalib/collecting/Collectors.py]: I don't think I can. 
The double-quote to single-quote conversion was performed by coala (Quote-Bear?), when I was running it in the repository to verify my changes.
The format also matches other messages, that need inline single-quotes (i.e. the message warning about unmatched bears in [Collectors.py Line 179](https://github.com/coala/coala/blob/master/coalib/collecting/Collectors.py).

I will amend the autoformatting issues introduced by my editor as soon as I can

Comment (SanketDG) [inline coalib/collecting/Collectors.py]: Oh, I see why `QuotesBear` did that. :+1:

File: coalib/collecting/Collectors.py [hunk 2]
```diff
@@ -82,7 +82,9 @@ def collect_files(file_paths, log_printer, ignored_file_paths=None,
         collected_files, file_globs_with_files = [], []
 
     _warn_if_unused_glob(log_printer, file_paths, file_globs_with_files,
-                         "No files matching '{}' were found.")
+                         "No files matching '{}' were found. "
```
Comment (Udayan12167) [inline coalib/collecting/Collectors.py]: Why can't you use single quotes here?

File: tests/collecting/CollectorsTest.py [hunk 3]
```diff
@@ -30,7 +30,10 @@ def test_file_invalid(self):
         self.assertEqual(collect_files(['invalid_path'],
                                        self.log_printer), [])
         self.assertEqual([log.message for log in self.log_printer.logs],
-                         ["No files matching 'invalid_path' were found."])
+                         ["No files matching 'invalid_path' were found. "
```
Comment (Udayan12167) [inline tests/collecting/CollectorsTest.py]: You can use single quotes here. By escaping the internal single quotes.

File: coalib/collecting/Collectors.py [hunk 4]
```diff
@@ -82,7 +82,9 @@ def collect_files(file_paths, log_printer, ignored_file_paths=None,
...
+                         'If this rule is not required, you can remove it '
+                         'from your .coafile to deactivate this warning.')
```
Comment (Udayan12167) [inline coalib/collecting/Collectors.py]: Add which argument to change in order to remove it from the coafile.

File: tests/collecting/CollectorsTest.py (hunk 3 above)
Comment (mrtes) [inline tests/collecting/CollectorsTest.py]: I see. 
I didn't even think about that. 
However, this would not be consistent to  [Line 177 of current Collectors.py](https://github.com/coala/coala/blob/master/coalib/collecting/Collectors.py).

So should I include escaped single quotes for both occasions, or leave it as it is?

File: coalib/collecting/Collectors.py (hunk 4 above)
Comment (mrtes) [inline coalib/collecting/Collectors.py]: Something along the line: "remove the rule 'files = ...' from your .coafile?
I'm not sure, if this would be more helpful, than the current iteration.

However, if I could add the currently executing section, the message would be much more useful.
This info is actually available at the function call-site ([Processing.py Line 358](https://github.com/coala/coala/blob/master/coalib/processes/Processing.py)), so a change to the signature of "collect_files" could make this info accessible.
The only other call sites are in test-cases, which would have to be adapted as well.

All in all, this would require some more changes, but I should be able to do it.

Comment (mrtes) [inline coalib/collecting/Collectors.py]: I have gone ahead and implemented it in this fashion i.e. the section that includes the missing glob is named in the warning.
Changes were quite minor, thanks to optional parameters, but in hindsight i should probably have moved that change into its own commit (in case i have to revert it).

Looking forward to feedback.

**Comment by sils** [comment]
ack https://github.com/coala/coala/pull/4154/commits/56f087e435f76538873732fa8fdd9619e2b9545e I really like this, esp the addition of the section name is very practical

**Comment by mrtes** [comment]
Is there something that I have to do after approval?

**Comment by sils** [comment]
@mrtes either you rebase manually or you wait until we figure out cherry pick bot out with @adtac and @yukiisbored

**Comment by mrtes** [comment]
I went with the manual route. I hope everything is alright :)

**Comment by sils** [comment]
reack https://github.com/coala/coala/pull/4154/commits/b1a0cdcc05d6048fb62c55747276b9579b53e728

**Comment by sils** [comment]
@rultor merge
